test/sql/partitionwise.sql

```diff
@@ -0,0 +1,191 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+-- Create a two dimensional hypertable
+CREATE TABLE hyper (time timestamptz, device int, temp float);
+SELECT * FROM create_hypertable('hyper', 'time', 'device', 2);
+
+-- Create a similar PostgreSQL partitioned table
+CREATE TABLE pg2dim (time timestamptz, device int, temp float) PARTITION BY HASH (device);
+CREATE TABLE pg2dim_h1 PARTITION OF pg2dim FOR VALUES WITH (MODULUS 2, REMAINDER 0) PARTITION BY RANGE(time);
+CREATE TABLE pg2dim_h2 PARTITION OF pg2dim FOR VALUES WITH (MODULUS 2, REMAINDER 1) PARTITION BY RANGE(time);
+CREATE TABLE pg2dim_h1_t1 PARTITION OF pg2dim_h1 FOR VALUES FROM ('2018-01-01 00:00') TO ('2018-09-01 00:00');
+CREATE TABLE pg2dim_h1_t2 PARTITION OF pg2dim_h1 FOR VALUES FROM ('2018-09-01 00:00') TO ('2018-12-01 00:00');
+CREATE TABLE pg2dim_h2_t1 PARTITION OF pg2dim_h2 FOR VALUES FROM ('2018-01-01 00:00') TO ('2018-09-01 00:00');
+CREATE TABLE pg2dim_h2_t2 PARTITION OF pg2dim_h2 FOR VALUES FROM ('2018-09-01 00:00') TO ('2018-12-01 00:00');
+
+-- Create a 1-dimensional partitioned table for comparison
+CREATE TABLE pg1dim (time timestamptz, device int, temp float) PARTITION BY HASH (device);
+CREATE TABLE pg1dim_h1 PARTITION OF pg1dim FOR VALUES WITH (MODULUS 2, REMAINDER 0);
+CREATE TABLE pg1dim_h2 PARTITION OF pg1dim FOR VALUES WITH (MODULUS 2, REMAINDER 1);
+
+INSERT INTO hyper VALUES
+       ('2018-02-19 13:01', 1, 2.3),
+       ('2018-02-19 13:02', 3, 3.1),
+       ('2018-10-19 13:01', 1, 7.6),
+       ('2018-10-19 13:02', 3, 9.0);
+
+INSERT INTO pg2dim VALUES
+       ('2018-02-19 13:01', 1, 2.3),
+       ('2018-02-19 13:02', 3, 3.1),
+       ('2018-10-19 13:01', 1, 7.6),
+       ('2018-10-19 13:02', 3, 9.0);
+
+INSERT INTO pg1dim VALUES
+       ('2018-02-19 13:01', 1, 2.3),
+       ('2018-02-19 13:02', 3, 3.1),
+       ('2018-10-19 13:01', 1, 7.6),
+       ('2018-10-19 13:02', 3, 9.0);
+
+SELECT * FROM test.show_subtables('hyper');
+
+SELECT * FROM pg2dim_h1_t1;
+SELECT * FROM pg2dim_h1_t2;
+SELECT * FROM pg2dim_h2_t1;
+SELECT * FROM pg2dim_h2_t2;
+
+
+-- Compare partitionwise aggreate enabled/disabled. First run queries
+-- on PG partitioned tables for reference.
```

Comment:
I'd like to see example queries with group by time_bucket even if the initial implementation does not enable partitionwise aggregation for those

Reply:
I will add, although it will be similar to `date_trunc`.